apps/web/vibes/soul/sections/forgot-password-section/forgot-password-form.tsx

```diff
@@ -1,21 +1,26 @@
 'use client';
 
-import { SubmissionResult, getFormProps, getInputProps, useForm } from '@conform-to/react';
+import { getFormProps, getInputProps, SubmissionResult, useForm } from '@conform-to/react';
 import { getZodConstraint, parseWithZod } from '@conform-to/zod';
-import { useActionState, useEffect } from 'react';
+import { useActionState } from 'react';
 import { useFormStatus } from 'react-dom';
 
+import { FormStatus } from '@/vibes/soul/form/form-status';
 import { Input } from '@/vibes/soul/form/input';
 import { Button } from '@/vibes/soul/primitives/button';
 
 import { schema } from './schema';
 
 type Action<State, Payload> = (state: Awaited<State>, payload: Payload) => State | Promise<State>;
 
-export type ForgotPasswordAction = Action<SubmissionResult | null, FormData>;
+export type ForgotPasswordAction = Action<
+  (SubmissionResult & { successMessage?: string }) | null,
```

Comment:
We should wrap the submission result instead of extending it

Reply:
Fixed.